Bulldozer.CSV/CSVComponent.cs

```diff
@@ -488,804 +836,2680 @@ public static int GetAttributeFieldType( string attributeTypeString )
-         *
-         * EntityAttribute.csv file must be run before this file
-         * or the attributes already exist in Rock DB.
-         *
-         * Columns already numbered from EntityAttribute file:
-         * private const int AttributeEntityTypeName = 0;
-         * private const int AttributeId = 1;
-         * private const int AttributeRockKey = 2;    Note: Attribute values that do not find match on AttributeId to FK or FID will attempt to match on the Attribute Key.
-         */
+        /// <summary>
+        /// Adds the group attributes.
+        /// </summary>
+        private int AddGroupAttributes()
+        {
+            if ( this.GroupTypeDict == null )
+            {
+                LoadGroupTypeDict();
+            }
+            if ( this.GroupAttributeDict == null )
+            {
+                LoadGroupAttributeDict();
+            }
 
-        private const int AttributeValueId = 3;             /* String | Int [Optional] */
-        private const int AttributeValueEntityId = 4;       /* String | Int [Required] */
-        private const int AttributeValue = 5;               /* String [Required] */
+            var invalidDefinedTypeAttributes = new List<string>();
 
-        #endregion EntityAttributeValue Constants
+            var rockContext = new RockContext();
+            var attributeService = new AttributeService( rockContext );
+            var attributeCategoryList = new CategoryService( rockContext ).Queryable().Where( a => a.EntityTypeId == AttributeEntityTypeId ).ToList();
+            var groupAttributesToProcess = this.GroupAttributeCsvList.Where( a => !this.GroupAttributeDict.ContainsKey( a.Key ) );
+            var newAttributes = 0;
 
-        #region ConnectionRequest Constants
+            // Add any Group Attributes to Rock that aren't in Rock yet
+            ReportProgress( 0, string.Format( "Creating {0} new Group Attributes...", groupAttributesToProcess.Count() ) );
+            foreach ( var attribute in groupAttributesToProcess )
+            {
+                attribute.Key = attribute.Key;
 
-        /*
-         * This is the definition of the csv format for the ConnectionRequest.csv file.
-         *
-         */
+                if ( !this.GroupAttributeDict.ContainsKey( attribute.Key ) )
+                {
+                    // the group attribute category targets the grouptype by name
+                    var newGroupAttribute = new Rock.Model.Attribute()
+                    {
+                        Key = attribute.Key,
+                        Name = attribute.Name,
+                        Guid = Guid.NewGuid(),
+                        EntityTypeId = GroupEntityTypeId,
+                        EntityTypeQualifierColumn = "GroupTypeId",
+                        EntityTypeQualifierValue = this.GroupTypeDict.Values.FirstOrDefault( gt => gt.Name.Equals( attribute.Category ) )?.Id.ToString(),
```

Comment:
Should we add GroupTypeId to group-attribute csv and use it here instead? Might also need to check existing export scripts to see if they even work properly with this.